pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -229,83 +230,73 @@ func shouldOrphanDependents(e event, accessor meta.Object) bool {
 			return false
 		}
 	}
-	return sets.NewString(accessor.GetFinalizers()...).Has(api.OrphanFinalizer)
+	finalizers := accessor.GetFinalizers()
+	for _, finalizer := range finalizers {
+		if finalizer == api.FinalizerOrphan {
+			return true
+		}
+	}
+	return false
 }
 
-func (gc *GarbageCollector) orhpanDependents(owner *node, dependents []*node, retries int) error {
+// dependents are copies of pointers to the owner's dependents, they don't need to be locked.
+func (gc *GarbageCollector) orhpanDependents(owner objectReference, dependents []*node) error {
 	var failedDependents []objectReference
+	var errors []error
 	for _, dependent := range dependents {
-		deleteOwnerRefPatch := fmt.Sprintf(`{"metadata":{"ownerReferences":[{"$patch":"delete","uid":"%s"}]}}`, owner.identity.UID)
-		patched := false
-		for count := 0; count < retries; count++ {
-			_, err := gc.patchObject(dependent.identity, []byte(deleteOwnerRefPatch))
-			// note that if the target ownerReference doesn't exist in the
-			// dependent, strategic merge patch will NOT return an error.
-			if err == nil {
-				patched = true
-				break
-			} else {
-				utilruntime.HandleError(fmt.Errorf("orphaning %s failed with %v", dependent.identity, err))
-			}
-		}
-		if !patched {
-			failedDependents = append(failedDependents, dependent.identity)
+		deleteOwnerRefPatch := fmt.Sprintf(`{"metadata":{"ownerReferences":[{"$patch":"delete","uid":"%s"}]}}`, owner.UID)
+		_, err := gc.patchObject(dependent.identity, []byte(deleteOwnerRefPatch))
+		// note that if the target ownerReference doesn't exist in the
+		// dependent, strategic merge patch will NOT return an error.
+		if err != nil {
```

Comment:
Two thoughts:
* Maybe you should ignore not found errors, in case the dependent has already been deleted?
* Can you set a UID precondition so you're sure to act on the right object?